draft-pauly-intarea-proxy-config-pvd.md

```diff
@@ -137,15 +137,40 @@ match the hostname of the HTTP proxy. The "prefixes" array SHOULD be empty by de
 # Enumerating proxies within a PvD {#proxy-enumeration}
 
 This document defines a new PvD Additional Information key, `proxies`, that
-is an array of strings that is a list of proxy URIs (or URI templates
-{{!URITEMPLATE=RFC6570}}) that are available as part of a PvD. The new
-key is registered in {{iana}}.
-
-The kind of proxy is implied by the URI scheme and any template variables.
-For example, since UDP proxying {{CONNECTUDP}} has the URI template variables
-`target_host` and `target_port`, the URI
-"https://proxy.example.org:4443/masque{?target_host,target_port}" implies
-that the proxy supports UDP proxying.
+is an array of dictionaries, where each dictionary in the array defines
+a single proxy that is available as part of the PvD (see {{proxies-key-iana}}).
+Each proxy is defined by a proxy protocol, a proxy URI (or URI template
+{{!URITEMPLATE=RFC6570}}), along with potentially other keys.
+
+This document defines two mandatory keys for the sub-dictionaries in the
+`proxies` array, `protocol` and `uri`. These keys, defined below, are
+the initial contents of the proxy information key registry
+({{proxy-info-iana}}). Other optional keys can be added to the dictionary
+to further define or restrict the use of a proxy. Clients that do not
+recognize or understand a key in a proxy sub-dictionary MUST ignore the entire
```

Comment:
Erroring out on unknown keys means it's impossible to extend the dictionary in a backwards-compatible way. In Babel we solved this by having a notion of [mandatory vs non-mandatory](https://www.rfc-editor.org/rfc/rfc8966.html#section-4.4). For example here one way to do this could be to have a prefix in the key that identifies whether it's mandatory or not, such as `!foo` is mandatory but `?bar` isn't. So if you see an unknown key the first character tells you if you can ignore or need to ignore the entire proxy definition.

Reply:
Yeah I think this is something to discuss / work on — potentially as a follow-up

Reply:
Fine by me, please file an issue before merging this though so we don't forget

Reply:
Filed https://github.com/tfpauly/privacy-proxy/issues/245